System/Posix/Resource.hsc

```diff
@@ -100,12 +103,16 @@ unpackRLimit :: CRLim -> ResourceLimit
 unpackRLimit (#const RLIM_INFINITY)  = ResourceLimitInfinity
 unpackRLimit other
 #if defined(RLIM_SAVED_MAX)
-    | ((#const RLIM_SAVED_MAX) :: CRLim) /= (#const RLIM_INFINITY) &&
-      other == (#const RLIM_SAVED_MAX) = ResourceLimitUnknown
+    | ((#const RLIM_SAVED_MAX) :: CRLim) /= (#const RLIM_INFINITY)
+    , other == (#const RLIM_SAVED_MAX)
```

Comment:
So GHC only sees redundant pattern match if you use `,` instead of `&&`?

How did you spot it then?

Reply:
No, the redundant pattern match happens in both cases.

I had changed the code here, as I was using `considerAccessible` on the second pattern match and it was more consistent with `,` rather than `&&`. I didn't change it back, but I can do so if you prefer.

Reply:
I think what is most important is that the unix release on hackage matches what GHC ships with.